.github/workflows/build.yml

```diff
@@ -188,27 +249,78 @@ jobs:
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run ntp-client:test
 
     - name: io-sim [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run io-sim:test
 
     - name: io-sim-classes [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run io-sim-classes:test
 
     - name: typed-protocols-examples [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run typed-protocols-examples:tests
 
     - name: network-mux [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run network-mux:test
 
     # issue: #1818
     - name: ourobors-network-framework [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run ouroboros-network-framework:test -- -p '$0 != "typed-protocols.Socket.socket send receive IPv4"' 
 
     - name: ouroboros-network [test]
       run: cabal --builddir="$CABAL_BUILDDIR" run ouroboros-network:test
 
+
+    # - name: ouroboros-consensus [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus
+
+    # - name: ouroboros-consensus-mock [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-mock
+
+    # - name: ouroboros-consensus-test [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-test
+
+    # - name: ouroboros-consensus-mock-test [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-mock-test
+
+    # - name: ouroboros-consensus-byron [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-byron
+
+    # - name: ouroboros-consensus-byronspec [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-byronspec
+
+    # - name: ouroboros-consensus-byron-test [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-byron-test
+
+    # - name: ouroboros-consensus-shelley [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-shelley
+
+    # - name: ouroboros-consensus-shelley-test [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-shelley-test
+
+    # - name: ouroboros-consensus-cardano [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-cardano
+
+    # - name: ouroboros-consensus-cardano-test [test]
+    #   if: matrix.subset == 'consensus'
+    #   run: cabal --builddir="$CABAL_BUILDDIR" test ouroboros-consensus-cardano-test
+
```

Comment:
Why are these disabled? Because they take too long?

Reply:
Yep.  Nearly 2 hours I believe.

Reply:
To see the test times see this draft PR: https://github.com/input-output-hk/ouroboros-network/pull/2831

Reply:
> To see the test times see this draft PR: #2831

They `consensus` cells of the matrix are both currently at ~220 minutes.

Reply:
The specification for Github runners is here: https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners

Reply:
> GitHub hosts Linux and Windows runners on Standard_DS2_v2 virtual machines in Microsoft Azure with the GitHub Actions runner application installed.

> For more information about the Standard_DS2_v2 machine resources, see "[Dv2 and DSv2-series](https://docs.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series)" in the Microsoft Azure documentation.

> DSv2-series sizes run on Intel® Xeon® Platinum 8272CL (Cascade Lake), Intel® Xeon® 8171M 2.1GHz (Skylake) or the the Intel® Xeon® E5-2673 v4 2.3 GHz (Broadwell) or the Intel® Xeon® E5-2673 v3 2.4 GHz (Haswell) processors with Intel Turbo Boost Technology 2.0 and use premium storage.

> Size | vCPU | Memory: GiB | Temp storage (SSD) GiB | Max data disks | Max cached and temp storage throughput: IOPS/MBps (cache size in GiB) | Max uncached disk throughput: IOPS/MBps | Max NICs | Expected network bandwidth (Mbps)
> -- | -- | -- | -- | -- | -- | -- | -- | --
> Standard_DS2_v2 | 2 | 7 | 14 | 8 | 8000/64 (86) | 6400/96 | 2 | 1500

Reply:
^^^ So the machine's CPU is relatively underpower, with 2 relatively meager cores.

I'll do an experiment to see what portion of the time is spent on my laptop VM (not too much bigger than the Azure machine) in the ThreadNet tests and report back here.

Reply:
I ran the "consensus" test suites then I ran just their ThreadNet components. It's not quite apples to apples, but it's 11m23s for the whole thing, and 4m41s for just the ThreadNet parts. so about 40% time in ThreadNet.

```
ouroboros-consensus ouroboros-consensus-mock ouroboros-consensus-test ouroboros-consensus-mock-test ouroboros-consensus-byron ouroboros-consensus-byronspec ouroboros-consensus-byron-test ouroboros-consensus-shelley ouroboros-consensus-shelley-test ouroboros-consensus-cardano ouroboros-consensus-cardano-test
```

I used a fixed arbitrary QC replay seed. And I ran one test suites at a time, just like the GH action does.

I figured this on my laptop's VM would be loosely comparable to the GH action Azure machine. But then, compared to the numbers on the PR 2831 you shared above: mine are many times faster. EG the consensus 8.10.2 build, at https://github.com/input-output-hk/ouroboros-network/pull/2831/checks?check_run_id=1642016940, took ~200 minutes total. The breakdown of major/suspected summands there is

> build deps 30m + ouroboros-consensus-mock-test [test] 20m + ouroboros-consensus-byron-test [test] 60m + ouroboros-consensus-cardano-test [test] 60m + ouroboros-consensus-shelley-test [test] 7m = 177 minutes

I was careful to build them before running them. I peeked on the detailed logs on that experiment's PR and they also seem to have already been built before running for those times I quoted just above.

So... I think disabling the ThreadNets might save about 35% of the exec time or so? But that's based on the assumption that the reduction would be a similar percentage.

What do you think @newhoggy? Would that make a worthwhile difference? The ThreadNet tests are a "last line of defense" in some sense, but we at least run them via the Hydra push and merge CI and nightly.

Reply:
@newhoggy Here's a simpler option: let's just remove all of these lines and replace it with a single comment saying that the consensus tests take too long on GitHub's Azure machines.

I think I can Approve after that.